src/wp-includes/class-wp-block-type.php

```diff
@@ -268,6 +277,7 @@ class WP_Block_Type {
 	 *     @type string|null   $textdomain               The translation textdomain.
 	 *     @type array[]       $styles                   Alternative block styles.
 	 *     @type array[]       $variations               Block variations.
+	 *     @type array|null    $selectors                Custom CSS selectors for theme.json style generation.
```

Comment:
Does `null` have any special meaning here? I'm still not sure why we set `null` in some places but map it to an empty object/array on the client. It might be related to some backward compatibility considerations in places where `null` doesn't have a special role like for `example` where it disables block preview in the inserter.

Reply:
I don't think there is a special meaning for `null` here. It was more to keep it inline with the `supports` property. 

For now, I've tweaked the checks for the `selectors` property in https://github.com/WordPress/gutenberg/pull/46496 and updated this PR so that the default is an empty array. Happy to change this again though if we decide otherwise.